client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -102,15 +105,25 @@ const StatsLocations: React.FC< StatsModuleLocationsProps > = ( { query, summary
 	const geoMode = GEO_MODES[ selectedOption ];
 	const title = optionLabels[ selectedOption ]?.selectLabel;
 
+	const { supportsLocationsStats: supportsLocationsStatsFeature } = useSelector( ( state ) =>
+		getEnvStatsFeatureSupportChecks( state, siteId )
+	);
+
 	// Main location data query
 	const {
-		data = [],
+		data: locationsViewsData = [],
 		isLoading: isRequestingData,
 		isError,
 	} = useLocationViewsQuery< StatsLocationViewsData >( siteId, geoMode, query, countryFilter, {
-		enabled: ! shouldGate,
+		enabled: ! shouldGate && supportsLocationsStatsFeature,
 	} );
 
+	const countriesViewsData = useSelector( ( state ) =>
+		getSiteStatsNormalizedData( state, siteId, statType, query )
+	) as [ id: number, label: string ];
+
+	const data = supportsLocationsStatsFeature ? locationsViewsData : countriesViewsData;
```

Comment:
Should we rename `countriesViewsData` to indicate that it comes from a legacy endpoint? We could try something like `legacyCountriesViewsData`. Also, I think we should consider adding a comment to clarify what this refers to. As a developer unfamiliar with why we have different endpoints, I think it could be beneficial.

Reply:
Nice suggestion. I've updated the variable name and added a comment explaining when the legacy endpoint is used.